roles/telemetry_graphing/tasks/cypress_test.yml

```diff
@@ -0,0 +1,87 @@
+- name: Install cypress and the dependencies
+  block:
+    - name: Install Node.js
+      ansible.builtin.dnf:
+        name: nodejs
+        state: present
+      become: true
+
+    - name: Install dependencies
+      ansible.builtin.dnf:
+        name:
+          - xorg-x11-server-Xvfb
+          - gtk3-devel
+          - nss
+          - alsa-lib
+        state: present
+      become: true
+
+- name: Create required dirs and files
+  block:
+    - name: Create cypress-tests directory
+      ansible.builtin.file:
+        path: "{{ cypress_tests_dir }}"
+        state: directory
+        mode: '0755'  
+
+    - name: Create cypress/integration directory
+      ansible.builtin.file:
+        path: "{{ cypress_integration_dir }}"
+        state: directory
+        mode: '0755'  
+    
+    - name: Add cypress.config.js
+      ansible.builtin.copy:
+        src: "{{ role_path }}/files/cypress.config.js"  
+        dest: "{{ cypress_tests_dir }}/cypress.config.js"
+        mode: '0644' 
+
+    - name: Add dashboard-openstack-cloud test
+      ansible.builtin.copy:
+        src: "{{ role_path }}/files/dashboard-openstack-cloud.js"  
+        dest: "{{ cypress_integration_dir }}/dashboard-openstack-cloud.js"
+        mode: '0644' 
+      
+    - name: Create screenshot logs dir
+      ansible.builtin.file:
+        path: "{{ screenshots_dir }}"
+        state: directory
+        mode: '0755'
+
+- name: Initialize npm project
+  ansible.builtin.command: npm init -y
+  args:
+    chdir: "{{ cypress_tests_dir }}"
+  changed_when: false
+
+- name: Install Cypress as a dev dependency
+  ansible.builtin.command: npm install cypress --save-dev
+  args:
+    chdir: "{{ cypress_tests_dir }}"
+  changed_when: false
+
+# RHOSO-12658 need to be updated
+- name: "[TEST] Run Cypress tests to validate dashboards exist RHOSO-12658"
+  ansible.builtin.shell: |
+      Xvfb :99 -ac &
+      export DISPLAY=:99
+      npx cypress run
+  args:
```

Comment:
If this functionality is being kept here, I suggest splitting this into multiple tasks, i.e. "install_cypress" and "run_cypress_test"

Reply:
this is not to install Cypress we already have another task to do so, but if I split it to multiple shell task it causing some problem in the scope of export. so I would say its better to kept that way